decidim-core/lib/decidim/nicknamizable.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Decidim
+  # This concern contains the logic related to nicknames.
+  module Nicknamizable
+    def nickname_length_range
+      (0...nickname_max_length)
+    end
+
+    def nickname_max_length
+      20
+    end
+
+    def nicknamize(name)
+      disambiguate(name.parameterize(separator: "_")[nickname_length_range])
+    end
+
+    private
+
+    def disambiguate(name)
+      candidate = name
+
+      2.step do |n|
```

Comment:
How will this behave if there are let's say 10 users with the same computed nickname? Will this really keep adding the correct index, so last user will have `my_name_10` as nickname?

Reply:
Yup, that's the idea.